src/main/java/org/openjdk/jextract/impl/TreeMaker.java

```diff
@@ -226,7 +231,17 @@ public Declaration.Function createFunction(Cursor c) {
     }
 
     public Declaration.Constant createMacro(Position pos, String name, Type type, Object value) {
-        return Declaration.constant(pos, name, value, type);
+        Declaration.Constant macro = Declaration.constant(pos, name, value, type);
+        String valueString = value.toString();
+        if (value instanceof String) {
+            // quote string literal
+            valueString = STR."\"\{valueString}\"";
+        } else if (Utils.isPointer(type)) {
+            // add pointer cast to make it look different from a numeric constant
+            valueString = STR."(\{valueString})";
```

Comment:
Shouldn't this be something like:

```
valueString = STR."(\{type}) \{valueString}";
```

It seems like the current code would result in things like this:

    #define FOO (42)

Reply:
I've fixed this by adding a cast to `void*` always. Note that `type.toString` doesn't return something that is similar to the C decl, so we can't really use that. (whether that's something we want to fix is a question for another PR I think)